LoopKitUI/CarbKit/DateAndDurationSteppableTableViewCell.swift

```diff
@@ -0,0 +1,99 @@
+//
+//  DateAndDurationSteppableTableViewCell.swift
+//  LoopKitUI
+//
+//  Copyright © 2018 LoopKit Authors. All rights reserved.
+//
+
+import UIKit
+
+public class DateAndDurationSteppableTableViewCell: DatePickerTableViewCell {
+
+    public weak var delegate: DatePickerTableViewCellDelegate?
+
+    @IBOutlet public weak var titleLabel: UILabel!
+
+    @IBOutlet public weak var dateLabel: UILabel! {
+        didSet {
+            dateLabel.textColor = .secondaryLabel
+        }
+    }
+    
+    @IBOutlet weak var incrementButton: UIButton!
+
+    @IBOutlet weak var decrementButton: UIButton! 
+    
+    public var timeStepSize: TimeInterval = .minutes(15)
+    
+    public override var isDatePickerHidden: Bool {
+        didSet {
+            dateLabel.textColor = isDatePickerHidden ? .secondaryLabel : window?.tintColor
+        }
+    }
+        
+    private lazy var durationFormatter: DateComponentsFormatter = {
+        let formatter = DateComponentsFormatter()
+
+        formatter.allowedUnits = [.hour, .minute]
+        formatter.unitsStyle = .short
+
+        return formatter
+    }()
+    
+    private lazy var dateFormatter: DateFormatter = {
+        let dateFormatter = DateFormatter()
+        dateFormatter.doesRelativeDateFormatting = true
+        dateFormatter.dateStyle = .none
+        dateFormatter.timeStyle = .short
+    
+        return dateFormatter
+    }()
+
+    public override func updateDateLabel() {
+        if dateFormatter.doesRelativeDateFormatting {
+            dateFormatter.dateStyle = Calendar.current.isDateInToday(date) ? .none : .short
+        }
+        
+        switch datePicker.datePickerMode {
+        case .countDownTimer:
+            dateLabel.text = durationFormatter.string(from: duration)
+        case .date:
```

Comment:
[nit] Consider collapsing these 3 cases into one statement (to reduce duplicated code).